pkg/kubectl/cmd/config/use_context.go

```diff
@@ -82,10 +84,16 @@ func (o *useContextOptions) complete(cmd *cobra.Command) bool {
 	return true
 }
 
-func (o useContextOptions) validate() error {
+func (o useContextOptions) validate(config *clientcmdapi.Config) error {
 	if len(o.contextName) == 0 {
 		return errors.New("You must specify a current-context")
 	}
 
-	return nil
+	for _, context := range config.Contexts {
+		if context.Cluster == o.contextName {
```

Comment:
> @deads2k how does kubectl determine which clientcmd api version it's using?

It's down under `Cluster`: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/client/clientcmd/api/types.go#L60